base_layer/transaction_components/src/transaction_components/wallet_output.rs

```
@@ -157,6 +166,40 @@ impl WalletOutput {
         }
     }
 
+    /// This will create a new wallet output and try and calculate the required script key and input stack to spend this
+    /// output, will return None if it cannot calculate the script key or input stack
+    #[allow(clippy::too_many_arguments)]
+    pub async fn new_imported<KM: TransactionKeyManagerInterface>(
+        value: MicroMinotari,
+        commitment_mask_key_id: TariKeyId,
+        memo: MemoField,
+        output: TransactionOutput,
+        key_manager: &KM,
+    ) -> Option<Self> {
+        let mut output = WalletOutput::new_with_rangeproof(
+            output.version,
+            value,
+            commitment_mask_key_id,
+            output.features,
+            output.script,
+            Default::default(),
+            Default::default(),
+            output.sender_offset_public_key,
+            output.metadata_signature,
+            0,
+            output.covenant,
+            output.encrypted_data,
+            output.minimum_value_promise,
+            output.proof,
+            memo,
+        );
+        let (input_data, script_key) = output.get_script_private_key_id(key_manager).await.ok()??;
```

here by calling `ok()` we are discarding error but that could make debugging easier to find out why it failed.